labeling/codebase/localcodebase.go

```diff
@@ -14,23 +15,39 @@ type LocalCodebase struct {
 func (c LocalCodebase) FindFileMatching(
 	predicate func(string) bool,
 	glob ...string,
-) (string, error) {
-	for _, g := range glob {
-		filesFound, err := filepath.Glob(filepath.Join(c.BasePath, g))
-
-		if err != nil {
-			continue
-		}
-
-		for _, path := range filesFound {
-			relPath, err := filepath.Rel(c.BasePath, path)
-			if predicate(relPath) {
-				return relPath, err
+) (foundPath string, err error) {
+	basePath := c.BasePath
+	if basePath == "" {
+		basePath = "."
+	}
+	err = filepath.WalkDir(
+		basePath,
+		func(path string, d fs.DirEntry, fileError error) error {
+			if fileError != nil {
+				return fileError
+			}
+			for _, g := range glob {
+				relPath, innerErr := filepath.Rel(c.BasePath, path)
+				if innerErr != nil {
+					panic("unreachable")
```

Comment:
I wonder if we should return the error instead of panicking here
```suggestion
					return innerErr
```

Reply:
Yeah, not sure why I left the `panic` in there. Thanks for catching!